project/XDEv11/README.md

```diff
@@ -0,0 +1,70 @@
+# Maximum Graph Matching Library (MGML)
+
+## Basic Information
+
+In graph theory, a matching is a set of non-adjacent edges in a graph,
+so the maximum matching problem is to find the maximum set in terms of
+cardinality or weight.
+
+## Problem to Solve
+
+There are different algorithms for different situations. This project
+is trying to implement some popular algorithms in C++ and to also
+provide a API in python. Here are the problems I am going to solve.
+
+* Maximum Cardinality Bipartite Matching
+	* Hopcroft-Karp Algorithm ($O(|E|\sqrt{|V|})$)
+
+* Maximum Weight Bipartite Matching
+	* Hungarian Algorithm ($O(|V|^3)$)
+
+* Maximum Cardinality Matching in arbitrary graphs
+	* Blossom Algorithm ($|V||E|\alpha(|V|)$)
+	* MV Matching Algorithm (**optional**)
+
+* Maximum Weight Matching in arbitrary graphs
+	* Blossom Algorithm (**optional**)
+
+## Prospective Users
+
+Anyone who needs the answer of maximum matching problem.
+
+## System Architecture
+
+The algorithms are implemented in C++ for fast runtime, and there are
+interfaces in C++ for sure and in python by pybind11.
+
+## API Description
+
+The input will consist of a graph represented in adjacency list or
```

Comment:
It will help your implementation by providing pseudo code or signature.

Reply:
I have added some function signatures.

Reply:
It's clear.  Thanks.